### PR TITLE
feat: add `InputManipulator` directive

### PR DESCRIPTION
**Pull request by jaulz**
- [X] Added or updated tests
- [ ] Documented user facing changes
- [X] Updated CHANGELOG.md

**Changes**

As mentioned in https://github.com/nuwave/lighthouse/issues/2474 it would be great if there would be a way to manipulate nested input values. Unfortunately, we cannot change the `ArgManipulator` in a non-breakable way because the `$parentType` is different and also there is no `$parentField`. Thus, I think we need a new directive interface that allows users to manipulate values in a similar way like it's possible with arguments today. What do you think?

**Breaking changes**

No breaking changes. The failing tests are unrelated as far as I can see.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ You can find and compare releases at the [GitHub release page](https://github.co
 
 ## Unreleased
 
+- Add `InputManipulator` directive https://github.com/nuwave/lighthouse/pull/2476
```
Comment (spawnia) [inline CHANGELOG.md:12]:
```suggestion
### Added

- Add `InputManipulator` directive interface https://github.com/nuwave/lighthouse/pull/2476
```

File: src/Schema/AST/ASTBuilder.php [hunk 2]
```diff
@@ -222,5 +224,25 @@ protected function applyArgManipulators(): void
                 }
             }
         }
+    }    
+    
+    /** Apply directives on input that can manipulate the AST. */
```
Comment (spawnia) [inline src/Schema/AST/ASTBuilder.php:229]:
```suggestion
    /** Apply directives on input fields that can manipulate the AST. */
```

File: src/Support/Contracts/InputManipulator.php [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<?php declare(strict_types=1);
+
+namespace Nuwave\Lighthouse\Support\Contracts;
+
+use GraphQL\Language\AST\InputObjectTypeDefinitionNode;
+use GraphQL\Language\AST\InputValueDefinitionNode;
+use Nuwave\Lighthouse\Schema\AST\DocumentAST;
+
+interface InputManipulator extends Directive
```
Comment (spawnia) [inline src/Support/Contracts/InputManipulator.php:9]: I think this should be called `InputFieldManipulator`

**Comment by spawnia** [comment]
I am not quite sure where to add this in the docs. `argument-directives.md` seems misleading, perhaps we need to rename it to `input-value-directives.md` to be more general, or split it into `argument-directives.md` and `input-field-directives.md`. Some directives can be both, so I think we will need `input-value-directives.md` anyways.

**Comment by jaulz** [comment]
Sure, I will work on this and the other PR soon since I am quite busy right now but I didn't forget it ;)

**Comment by jaulz** [comment]
@spawnia I gave it a try, could you please have a look?

**Comment by jaulz** [comment]
@spawnia thanks!

**Pull request merged**